### PR TITLE
Remove old code

### PR DESCRIPTION
**Pull request by Anderman**
Removed some unused code

### CODE REVIEW

File: src/Microsoft.AspNet.Server.Kestrel/Http/SocketInput.cs [hunk 1]
```diff
@@ -17,7 +16,6 @@ public class SocketInput : ICriticalNotifyCompletion
 
         private readonly MemoryPool2 _memory;
         private readonly IThreadPool _threadPool;
-        private readonly ManualResetEventSlim _manualResetEvent = new ManualResetEventSlim(false);
```
Comment (halter73) [inline src/Microsoft.AspNet.Server.Kestrel/Http/SocketInput.cs]: Removing this seems scary. I think this breaks reasonable assumptions about how GetResult() works.

**Comment by Anderman** [comment]
Yeah, little bit scary. But I can't think of any situation when this is needed. And there is no test or comment why this is needed. 

The code is created by @lodejard

**Comment by halter73** [comment]
I'm happy taking the first commit removing uncalled methods, but I don't see a compelling enough reason to remove `_manualResetEvent` from `SocketInput`.

**Comment by cesarbs** [comment]
Same for me.

**Comment by Anderman** [comment]
Ok, do you want me to remove the last commit?

**Comment by cesarbs** [comment]
Yes, please.

**Comment by halter73** [comment]
SocketInput's Skip and Take methods have been removed #525, and I don't want to remove the `_manualResetEvent`, so I'm closing this.

**Pull request closed without merging**